src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj

```diff
@@ -11,17 +11,14 @@
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <!-- Set this to false because assemblies should not reference this assembly directly, (except for tests, of course). -->
     <IsProjectReferenceProvider>false</IsProjectReferenceProvider>
+    <UseAspNetCoreSharedRuntime>true</UseAspNetCoreSharedRuntime>
+    <DoNotApplyWorkaroundsToMicrosoftAspNetCoreApp>true</DoNotApplyWorkaroundsToMicrosoftAspNetCoreApp>
```

Comment:
Is this really what we want for the existing place we're doing this (Microsoft.AspNetCore.Watch.BrowserRefresh.csproj❔ And the two places you're extending the hack❔

The problem is these packages are now building against whatever version of the Microsoft.AspNetCore.App shared Fx is in the SDK. We don't update the SDK often enough to be sure these projects build and run correctly.

Suggest instead that we scrap the `$(DoNotApplyWorkaroundsToMicrosoftAspNetCoreApp)` workaround and instead ensure the `$(SharedFxVersion)` version of Microsoft.AspNetCore.App.Runtime (and, if necessary, Microsoft.AspNetCore.App.Ref) has built before these three projects. The block at https://github.com/dotnet/aspnetcore/blob/ef1f48ab97174131f1753efe9241295b1309d3e2/src/Framework/test/Microsoft.AspNetCore.App.UnitTests.csproj#L54-L63 does this and could be copied around or centralized.

Reply:
I didn't try exactly this, but something similar. In the absence of a FrameworkReference all of the shared fx assemblies become part of the publish output. How do I resolve that?